developer_documentation/contribution_guidelines.Rmd

```diff
@@ -91,6 +92,74 @@ the devil].
+    bypassed.
+
+  - All pull requests into `main` and `develop` require approval before merging.
+    This requirement is not enforced using GitHub branch protection rules, but
+    please do not unilaterally merge a pull request without any form of approval
+    from another developer. (There are two exceptions to this rule related to
+    hotfix and release branches, described below.)
+
+  - The BioCro R package and repository follow semantic versioning of the form
+    `major.minor.bug`. Hotfix branches should typically only increment the third
+    (`bug`) number. Most release branches will increment the second (`minor`)
+    number, and very rarely the first (`major`) number. Feature branches do not
+    directly change the version number.
+
+  - Branch names should be prefixed with the branch type: either `feature/`,
+    `hotfix/`, or `release/`. Hotfix and release branches should be named with
+    the new version number; for example, `hotfix/v3.0.3` or `release/v3.1.0`.
+    Feature branch names should reflect their purpose, such as
+    `feature/describe-branch-rules`.
+
+  - `NEWS.md` should be updated whenever the package version changes.
+
+* The following is a short description of this branching model:
+
+  - The `main` branch always contains the latest stable release of the BioCro
+    GitHub repository. In fact, a new tag and release is made any time changes
+    are merged into `main`, and such changes should always be accompanied by an
+    increment to the BioCro R package version number.
+
+  - The `develop` branch contains new BioCro features that are under development
+    but not yet ready for a new release.
+
+  - Hotfix branches are for relatively small changes that are nevertheless
+    important enough to warrant immediate incorporation into `main`. Examples
+    include bug fixes and crucial changes to GitHub workflows. A hotfix branch
+    should be based off `main` and first merged into `main` via an approved pull
+    request. Then, a second pull request should be made to merge into `develop`.
+    If there are no merge conflicts or test failures, this request can be merged
+    without any additional approval. When both merges are complete, the hotfix
+    branch should be deleted.
+
+  - Feature branches are for more complex changes that do not require immediate
+    release; for example, the addition of a new R function to the package
+    namespace. A feature branch should be based off `develop` and merged back
+    into `develop`. When the merge into `develop` is complete, the feature
+    branch should be deleted.
+
+  - When sufficient changes have accumulated in `develop` to justify a new
+    version of the package, a release branch is used to move the changes from
+    `develop` to `main`. Release branches should not include substantive
+    changes; rather, a release branch is primarily used to increment the package
+    version and to ensure that the changelog in `NEWS.md` is up-to-date. A
```

Comment:
I made a change to the text, but not 100% I interpreted this comment the right way.

Reply:
When there's a compound modifier, it is can ne hyphenated and sometimes it is necessary to prevent confusion. But if the words aren't being used as a compound modifier, they are never hyphenated. So you could say "an up-to-date changelog" (up-to-date acts as a single adjective) or "the changelog is up to date" (up to date is not an adjective here).